api/src/routers/shop.py

```diff
@@ -0,0 +1,73 @@
+from typing import Any
+
+from fastapi import APIRouter, Depends, HTTPException
+from sqlmodel import Session
+from src.classes.company import CompanyRepresentation
+from src.classes.pagination import ShopPagination
+from src.classes.shop import ShopRepresentation
+from src.db import get_session
+from src.models import ShopItemCreate, ShopItemPublic, ShopItemPurchase, ShopItemUpdate
+
+router = APIRouter()
+
+
+@router.get("/shop")
+async def get_shop(params: ShopPagination = Depends(), session: Session = Depends(get_session)) -> dict[str, Any]:
+    """Get the items that are currently available in the shop to purchase.
+
+    :param params: Pagination parameters.
+    :param session: Database session.
+    :return: Fetched shop items
+    """
+    return ShopRepresentation.fetch_shop(session=session, params=params)
+
+
+@router.post("/shop")
+async def create_shop_item(data: ShopItemCreate, session: Session = Depends(get_session)) -> None:
+    """Create a new Item in the Shop.
+
+    :param data: Shop Item data.
+    :param session: Database session.
+    :return: None
+    """
+    ShopRepresentation.create_item(session=session, data=data)
+    raise HTTPException(status_code=200, detail="Shop Item successfully created.")
```

Comment:
It's a little odd to raise an Exception on a success, maybe returning a string would work or a `Response` object